src/core.js

```diff
@@ -392,6 +393,99 @@ define([
         ];
     };
 
+    fn._init_bulk_update_questions = function () {
+        var _this = this,
+            menuItems = this.getBulkUpdateMenuItems();
+
+        var $lastItem = this.$f.find('.fd-bulk-update-header');
+        $lastItem.nextUntil(".divider").remove();
+        _(menuItems).each(function (menuItem) {
+            var $menuLink = $(util.format(
+                    "<a tabindex='-1' href='#'>{name}</a>",
+                    _.extend(menuItem)
+                )).click(function (e) {
+                    e.preventDefault();
+                    _this.showConfirmBulkActionModal(menuItem.confirmMessage, menuItem.confirmAction);
+                }),
+                $newItem = $("<li></li>").append($menuLink);
+            $lastItem.after($newItem);
+            $lastItem = $newItem;
+        });
+
+    };
+
+    fn.getBulkUpdateMenuItems = function () {
+        var _this = this,
+            menuItems =  [
+                {
+                    name: gettext('Make Required'),
+                    confirmMessage: gettext("You are about to make all existing " +
+                        "questions on this form required. This action will overwrite " +
+                        "the current configuration of this form and can only be " +
+                        "undone by editing individual questions."),
+                    confirmAction: function () {
+                        _this.makeAllQuestionsRequired();
+                    },
+                },
+            ];
+
+        if (_this.data.core.databrowser) {
+            menuItems.push({
+                name: gettext('Load Default Values'),
+                confirmMessage: gettext("You are about to set matching case " +
+                    "properties as Default Values for all existing questions. " +
+                    "This action will overwrite existing Default Values for " +
+                    "all questions with matching case properties and can only " +
+                    "be undone by editing individual questions."),
+                confirmAction: function () {
+                    _this.defaultMatchingQuestionsToCaseProperties();
+                },
+            });
+        }
+        return menuItems;
+    };
+
+    fn.makeAllQuestionsRequired = function () {
+        var _this = this;
+        _this.data.core.form.walkMugs(function (mug) {
+            if (mug.spec.requiredAttr && mug.isVisible('requiredAttr')) {
```

Comment:
It probably makes sense to move the `mug.spec.requiredAttr` (more generally `mug.spec[property]`) check into `mug.isVisible()`. I'm surprised it doesn't already check for that, but it should be fine to make it return false immediately if there is no spec for the given property.

Reply:
hey, so I looked into this, and I think I don't want to do that because `isVisible` first checks `getPresence` which explicitly throws an error when the property is not present:
https://github.com/dimagi/Vellum/blob/4557be6d99e1bb5c081a83a0f57bf3bd077cbebb/src/mugs.js#L329-L331

I want this change to be as isolated as possible so modifying existing code when there is already a check in place for the opposite seems like a bad idea.